pkg/aws/client/client.go

```diff
@@ -128,6 +129,80 @@ func (c *Client) GetVPCInternetGateway(ctx context.Context, vpcID string) (strin
 	return "", nil
 }
 
+// GetElasticIPsAssociationIDForAllocationIDs list existing elastic IP addresses for the given allocationIDs.
+// returns a map[elasticIPAllocationID]elasticIPAssociationID or an error
+func (c *Client) GetElasticIPsAssociationIDForAllocationIDs(ctx context.Context, allocationIDs []string) (map[string]*string, error) {
+	describeAddressesInput := &ec2.DescribeAddressesInput{
+		AllocationIds: aws.StringSlice(allocationIDs),
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("domain"),
+				Values: aws.StringSlice([]string{"vpc"}),
+			},
+		},
+	}
+
+	describeAddressesOutput, err := c.EC2.DescribeAddressesWithContext(ctx, describeAddressesInput)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(describeAddressesOutput.Addresses) == 0 {
+		return nil, nil
+	}
+
+	result := make(map[string]*string, len(describeAddressesOutput.Addresses))
+	for _, addr := range describeAddressesOutput.Addresses {
+		if addr.AllocationId == nil {
+			// this should not happen
```

Comment:
If it "should" not happen, why do we have this code? :) I would remove this comment